### DISCUSSION TITLE
Allow "Buy Now" functionality on NFTs with unlockables

### DISCUSSION THREAD

**Discussion by sungkhum** [deso-protocol/dips#203] [Medium CIPs]
Currently if an NFT has an unlockable it can not be sold with a "Buy Now" option. It would be great if the seller could enter in the unlockable content before it is sold and still be able to select to sell it with the "Buy Now" option.

Use cases are numerous. I use the unlockable to store NFT rarity data (hidden until user buys). Also many use the unlockable to store urls to additional downloads/websites. Nothing from a user perspective prevents the seller from entering the data in the unlockable before it is sold instead of the way it has been with auction NFT sales, where the unlockable data is entered when the NFT is sold.

**Comment by starykolarz** [comment]
As buyer is unknown at moment of minting and NFT, it's not possible to encrypt unclockable content when minting.

Solutions from top of my head with various levels of usability and effort required, in no particular order:
1. Implementation of an acceptance process of buy it now purchase, to allow the seller to encrypt content for specific buyer, after the buyer uses buy now option.
2. Creation of an escrow account keeping the unlockable content to transfer to seller (maybe this is an idea for a paid service?) - without changes, it could be done via DMs, could be automated. Cumbersome.
3. Creator informs about delivering additional high quailty content (the "unlockable") in NFT post body; then her/his reputation depends on delivering on that promise via DM after buy now.

I think option 3 is the one to go with for now.

**Reply by sungkhum** [reply]
Ah, right - didn't think about the fact that the user has to be known to encrypt...oh well...
